static/app/components/modals/addDashboardIssueWidgetModal.tsx

```diff
@@ -107,13 +108,21 @@ class AddDashboardIssueWidgetModal extends React.Component<Props, State> {
     } = this.props;
     this.setState({loading: true});
     let errors: FlatValidationError = {};
+
+    // Clear any empty fields
+    const queries = this.state.queries.map(query => {
+      return {...query, fields: query.fields.filter(field => field)};
+    });
+    this.setState({queries});
```

Comment:
hmm I think because of this, I can add a table with no columns.
1. First column has a field `issue`
2. Add a column but don't select a field
3. Delete the first `issue` column
4. Save

Reply:
Nice catch, I added a new condition here to attach an error to the query `fields` when no columns are selected.